src/main/scala/io/iohk/ethereum/crypto/EthereumIESEngine.scala

```diff
@@ -111,7 +101,7 @@ class EthereumIESEngine(kdf: Either[ConcatKDFBytesGenerator, MGF1BytesGeneratorE
         val derivedKey: ByteString = fillKDFunction(ECIESCoder.KeySize / 8 + ECIESCoder.KeySize / 8)
         val (firstPart, secondPart) = derivedKey.splitAt(ECIESCoder.KeySize / 8)
 
-        if (IV != null)
+        if (Option(IV).isDefined)
```

Comment:
@whysoserious it can be null because IV vector may be not provided to 
https://github.com/input-output-hk/etc-client/blob/phase/2/txHashValidation/src/main/scala/io/iohk/ethereum/crypto/ECIESCoder.scala#L36
but as I see this method is used only in one place now so we can remove this check or replace it with Option
`IV: Option[Array[Byte]]`